tools/clang/lib/Sema/SemaExprMember.cpp

```diff
@@ -1762,12 +1768,17 @@ Sema::BuildImplicitMemberExpr(const CXXScopeSpec &SS,
     if (SS.getRange().isValid())
       Loc = SS.getRange().getBegin();
     CheckCXXThisCapture(Loc);
-    baseExpr = new (Context) CXXThisExpr(loc, ThisTy, /*isImplicit=*/true);
+    if (getLangOpts().HLSL && ThisTy->isPointerType()) {
+      baseExpr = genereateHLSLThis(Loc, ThisTy, /*isImplicit=*/true);
+      ThisTy = ThisTy->getAs<PointerType>()->getPointeeType();
+    } else
+      baseExpr = new (Context) CXXThisExpr(loc, ThisTy, /*isImplicit=*/true);
```

Comment:
This leaves just a couple places where we create a new `CXXThisExpr` without calling `generateHLSLThis()` when the language is HLSL and it's a pointer. Should we do it everywhere?

Reply:
I was hesitant to change the code in code paths that I wasn't sure were reachable. Like lambda captures. I don't particularly like making changes that I can't write tests for.

Reply:
That's reasonable. The only one that I found not involved in lambdas is this: https://github.com/microsoft/DirectXShaderCompiler/blob/39dd31f3552632de480fcc575de452f673915737/tools/clang/lib/Sema/SemaExpr.cpp#L1900

Not sure exactly how to get there either though and it is for a failure diagnostic, so it's pretty minor.